### PR TITLE
fix linting / tsc smaller ci fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
This MR attempts to get CI to pass. Turns out the images are quite old so I need to update dependencies on there so i can get newer java/node/etc on there so android can actually build. This MR fixes the tsc/lint issues. Let's fix the other parts in later MRs

### CODE REVIEW

File: shared/actions/platform-specific/index.native.tsx [hunk 1]
```diff
@@ -423,7 +424,7 @@ const handleFilePickerError = (action: ConfigGen.FilePickerErrorPayload) => {
 const editAvatar = async () => {
   try {
     const result = await launchImageLibraryAsync('photo')
-    return result.cancelled === true
+    return result.cancelled
```
Comment (songgao-zoom) [inline shared/actions/platform-specific/index.native.tsx:427]: this is different if cancelled is null/undefined

File: shared/actions/settings.tsx [hunk 2]
```diff
@@ -114,7 +114,7 @@ const toggleNotifications = async (state: Container.TypedState) => {
   )
 
   if (!result || !result.body || JSON.parse(result.body)?.status?.code !== 0) {
-    throw new Error(`Invalid response ${result || '(no result)'}`)
+    throw new Error(`Invalid response ${result.body || '(no result)'}`)
```
Comment (songgao-zoom) [inline shared/actions/settings.tsx:117]: do we need to test `result` not being null/undefined?

```suggestion
    throw new Error(`Invalid response ${result?.body || '(no result)'}`)
```

**Review by songgao-zoom — APPROVED** [review]

File: shared/actions/platform-specific/index.native.tsx (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/platform-specific/index.native.tsx:427]: this is typed as boolean. the new linting is TS aware

File: shared/actions/settings.tsx (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/settings.tsx:117]: https://github.com/keybase/client/pull/24877
